src/constructs/core/mappings.ts

```diff
@@ -1,13 +1,13 @@
 import { CfnMapping } from "@aws-cdk/core";
-import type { Stage } from "../../constants";
+import type { Stage, StageForInfrastructure } from "../../constants";
 import type { AppIdentity } from "./identity";
 import type { GuStack } from "./stack";
 
 export type GuMappingValue = string | number | boolean;
 
 export interface GuStageMappingValue<T extends GuMappingValue> extends AppIdentity {
   variableName: string;
-  stageValues: Record<Stage, T>;
+  stageValues: Record<Stage, T> | Record<StageForInfrastructure, T>;
```

Comment:
`stageValues` can be:

```typescript
{
  "CODE": 1,
  "PROD": 2,
}
```

or

```typescript
{
  "INFRA": 1
}
```

That is, via the type system, stage is either `CODE or PROD` or `INFRA`.